### PR TITLE
Algorithms-side Schedule Object Models

### PR DESCRIPTION
**Pull request by rbassot**
Completes the Schedule object within ticket #62.

Five separate class models were created to build the full nested Schedule object, as defined by the Algorithms teams. Additions in this PR include:
- 5 class models set with primary keys (where applicable, as documented in the models file), and configured with necessary dependencies with eachother class. Each model having custom `__str__` methods defined to increase usability on the Django Admin page
- 5 serializers, nested so the exterior ScheduleSerializer() can build an entire model at once
- A test script that builds an entire Schedule object from the ground up, into a format that is serializable into Python dicts or JSON nested object

Kindly assure that the test script builds a Schedule object as expected, that the models are logically correct, and that the same can be done from the Django Admin page.

-----
`serializer.data` should be a traversable Python dictionary, as seen below:

image

-----
Sample JSON-converted output from the serializer script (update to fix `professor: id --> int`):
```
{
	"fall": [{
		"course": {
			"code": "CSC226",
			"title": "Algorithms 1",
			"pengRequired": {
				"fall": true,
				"spring": false,
				"summer": true
			},
			"yearRequired": 2
		},
		"sections": [{
			"professor": {
				"id": 1,
				"name": "Mike Zastre"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "MONDAY",
				"timeRange": ["12:00", "13:00"]
			}, {
				"dayOfWeek": "TUESDAY",
				"timeRange": ["13:00", "14:00"]
			}]
		}, {
			"professor": {
				"id": 2,
				"name": "Bill Bird"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "WEDNESDAY",
				"timeRange": ["11:00", "12:00"]
			}, {
				"dayOfWeek": "THURSDAY",
				"timeRange": ["13:00", "15:00"]
			}]
		}]
	}, {
		"course": {
			"code": "SENG275",
			"title": "Testing 1",
			"pengRequired": {
				"fall": false,
				"spring": true,
				"summer": true
			},
			"yearRequired": 2
		},
		"sections": [{
			"professor": {
				"id": 2,
				"name": "Bill Bird"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "WEDNESDAY",
				"timeRange": ["11:00", "12:00"]
			}, {
				"dayOfWeek": "THURSDAY",
				"timeRange": ["13:00", "15:00"]
			}]
		}]
	}],
	"spring": [{
		"course": {
			"code": "SENG275",
			"title": "Testing 1",
			"pengRequired": {
				"fall": false,
				"spring": true,
				"summer": true
			},
			"yearRequired": 2
		},
		"sections": [{
			"professor": {
				"id": 2,
				"name": "Bill Bird"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "WEDNESDAY",
				"timeRange": ["11:00", "12:00"]
			}, {
				"dayOfWeek": "THURSDAY",
				"timeRange": ["13:00", "15:00"]
			}]
		}]
	}],
	"summer": [{
		"course": {
			"code": "CSC226",
			"title": "Algorithms 1",
			"pengRequired": {
				"fall": true,
				"spring": false,
				"summer": true
			},
			"yearRequired": 2
		},
		"sections": [{
			"professor": {
				"id": 1,
				"name": "Mike Zastre"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "MONDAY",
				"timeRange": ["12:00", "13:00"]
			}, {
				"dayOfWeek": "TUESDAY",
				"timeRange": ["13:00", "14:00"]
			}]
		}, {
			"professor": {
				"id": 2,
				"name": "Bill Bird"
			},
			"capacity": 0,
			"timeSlots": [{
				"dayOfWeek": "WEDNESDAY",
				"timeRange": ["11:00", "12:00"]
			}, {
				"dayOfWeek": "THURSDAY",
				"timeRange": ["13:00", "15:00"]
			}]
		}]
	}]
}
```

Unit test suites will be added in a later update.

### CODE REVIEW

File: schedule/Schedule_models.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from django.db import models
+from django.contrib.postgres.fields import ArrayField
+from itertools import chain
+from django.utils.translation import gettext_lazy as _
+
+
+'''PRIMARY KEY: dayOfWeek + timeRange'''
+class A_TimeSlot(models.Model):
```
Comment (Russell-Waterhouse) [inline schedule/Schedule_models.py:8]: the A_ prefix just indicates this is for algorithms correct?

Comment (rbassot) [reply schedule/Schedule_models.py:8]: That's right, I thought it was a clean naming convention to avoid conflicting with FE-side models, like `Course`

**Review by Russell-Waterhouse — APPROVED** [review]
This looks awesome! Thanks for all your hard work on this!

**Comment by rbassot** [comment]
Fixed an inconsistency noticed by Shea that the `professor.id` should be an integer type. Models are unaffected due to the use of JSONField, but the build script provided was adjusted.

**Pull request merged**